core/vdbe/execute.rs

```diff
@@ -6832,6 +6832,11 @@ pub fn op_create_btree(
     Ok(InsnFunctionStepResult::Step)
 }
 
+pub enum OpDestroyState {
+    CreateCursor,
+    DestroyBtree(Arc<RwLock<BTreeCursor>>),
```

Comment:
there's really no other reason for this to be `Arc<RwLock<T>>` except that it needs to be mutable and ideally heap allocated so the enum isn't huge